### PR TITLE
chore: bump SOR to 4.7.2 - fix: add v4 related split routes metrics and routeToString

### PR DESCRIPTION
**Pull request by jsy1218**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

- **What is the current behavior?** (You can also link to an open issue here)
forgot to emit v4 related split routes metrics

- **What is the new behavior (if this is a feature change)?**
emit v4 related split routes metrics

- **Other information**:
this is to keep track of an edge case, that i dont expect happen frequently https://linear.app/uniswap/issue/ROUTE-303/tech-debt-split-route-can-have-different-ethweth-input-or-output#comment-a0b9a66a

### CODE REVIEW

**Comment by jsy1218** [comment]
* **#750** <a href="https://app.graphite.dev/github/pr/Uniswap/smart-order-router/750?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jsy1218 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: package.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "@uniswap/smart-order-router",
+  // TODO: bump patch version
```
Comment (cgkol) [inline package.json:3]: you probably still need that here right? happy to review when reayd

Comment (jsy1218) [reply package.json:3]: will let you know

Comment (jsy1218) [reply package.json]: done

File: src/util/routes.ts [hunk 2]
```diff
@@ -50,8 +50,8 @@ export const poolToString = (pool: TPool): string => {
       pool.token0,
       pool.token1,
       pool.fee,
-      0,
-      ADDRESS_ZERO
+      pool.tickSpacing,
```
Comment (jsy1218) [inline src/util/routes.ts:53]: this is only being used in the quote response routeToString field:

```
{
    "requestId": "d4fbfbd2-1abf-42a9-b87a-5ff193d47666",
    "routing": "CLASSIC",
    "quote": {
        "chainId": 11155111,
        "input": {
            "amount": "14251349",
            "token": "0x0000000000000000000000000000000000000000"
        },
        "output": {
            "amount": "12000000",
            "token": "0x1c7d4b196cb0c7b01d743fbc6116a902379c7238",
            "recipient": "0xE49ACc3B16c097ec88Dc9352CE4Cd57aB7e35B95"
        },
        "swapper": "0xE49ACc3B16c097ec88Dc9352CE4Cd57aB7e35B95",
        "route": [
            [
                {
                    "type": "v4-pool",
                    "address": "0x31a0d134cae16b840bfea2cde2c18328024a345bb8459381b5c747ee80552280",
                    "tokenIn": {
                        "chainId": 11155111,
                        "decimals": "18",
                        "address": "0x0000000000000000000000000000000000000000",
                        "symbol": "ETH"
                    },
                    "tokenOut": {
                        "chainId": 11155111,
                        "decimals": "6",
                        "address": "0x1c7D4B196Cb0C7B01d743Fbc6116a902379C7238",
                        "symbol": "USDC"
                    },
                    "fee": "1500",
                    "tickSpacing": "40",
                    "hooks": "0x0000000000000000000000000000000000000000",
                    "liquidity": "85647129",
                    "sqrtRatioX96": "78517620563033212495166708499",
                    "tickCurrent": "-181",
                    "amountIn": "14251349",
                    "amountOut": "12000000"
                }
            ]
        ],
        "slippage": 0.5,
        "tradeType": "EXACT_OUTPUT",
        "quoteId": "2066db21-6517-432f-a5bb-d0bc3b45842a",
        "gasFeeUSD": "0.000696242491458797",
        "gasFeeQuote": "1189760694576375",
        "gasUseEstimate": "131325",
        "priceImpact": 14.27,
        "txFailureReasons": [],
        "maxPriorityFeePerGas": "1000000000",
        "maxFeePerGas": "11423687737",
        "gasFee": "1500215792061525",
        "routeString": "[V4] 100.00% = ETH -- 0.15% [0xb793c6019fbdb95575103e55874d923da9386f689076f64659f1fff6bd3ac31f]USDC",
        "blockNumber": "6887438"
    },
    "permitData": null
}
```

routeToString doesn't impact any business logics

**Review by xrsv — APPROVED** [review]

**Comment by jsy1218** [comment]
### Merge activity

* **Oct 16, 1:36 PM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/Uniswap/smart-order-router/750).

**Pull request merged**